radicale_imap/__init__.py

```diff
@@ -32,16 +32,44 @@ class Auth(BaseAuth):
     type = radicale_imap
     imap_host = imap.server.tld
     imap_secure = True
+    imaps = True # for "old"-style imap over tls
 
     """
 
+    def get_connection(self, host, port):
+        return imaplib.IMAP4(host=host, port=port)
+
+    def get_secure_connection(self, host, port, imaps):
+        try:
+            if imaps:
+                connection = imaplib.IMAP4_SSL(host=host, port=port)
```

Comment:
The hostname is not checked. This is very insecure.
Enable the check if ``secure`` is true.